app/controllers/admin/applications_controller.rb

```diff
@@ -0,0 +1,28 @@
+class Admin::ApplicationsController < ApplicationController
+    def show
+        @application = Application.find(params[:id])
+        
+        @pets = Pet.search(params[:search])
+        
+        if params[:added_pet].present? 
+            pet_id = params[:added_pet]
+            @application.add_pet_to_application(pet_id)
+        end
+
+        if params[:pet_ownership_description].present?
```

Comment:
This functionality can also just live in the applications#show rather than the admin/applications#show!